src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaClientProduceFactory.java

```diff
@@ -1731,11 +1736,58 @@ private void doEncodeProduceRequest(
             final CRC32C crc = crc32c;
             crc.reset();
             crc.update(encodeSlotByteBuffer);
-            encodeSlotBuffer.putInt(encodeSlotOffset + crcOffset, (int) crc.getValue(), BIG_ENDIAN);
 
+            long checksum = crc.getValue();
+            if (dataFlags != FLAGS_FIN)
+            {
+                checksum = getChecksum(encodeBuffer, encodeLimit, encodeProgress, encodeSlotBuffer, checksum);
+            }
```

Comment:
Wouldn't we want to calculate the checksum the same way whether or not `fin` flag is present?